src/index.js

```diff
@@ -25,8 +25,10 @@ export default (block, options = {}) => (props) => {
     .filter(a => a)
     .reduce((a, b) => a.concat(b), [])
 
-  const classNames = makers.map(maker => maker(blockName, props, { delimiters }))
+  const classNamesList = makers.map(maker => maker(blockName, props, { delimiters }))
+  const className = cx(classNamesList).trim().replace(/\s\s+/g, ' ')
```

Comment:
@paranoida  you're right
the problem here is you change 1 whitespace with whitespace all the time, which is not really efficient, I guess